### DISCUSSION TITLE
CN name after TLS termination on LB

### DISCUSSION THREAD

**Discussion by mntnclmbr** [emqx/emqx#9945] [Q&A] [answered]
Hi,

If we terminate TLS on the loadbalancer, can we still get the CN name and use it as the username ? My understanding is for this to work, TLS has to be terminated on the node itself.

**Comment by id** [comment, marked as answer]
@mntnclmbr yes, if your loadbalancer support proxy protocol v2.

See, for example, this blog post https://tealfeed.com/haproxy-tls-termination-client-common-name-lciyk, and the test script https://github.com/zmstone/emqx-haporxy-example/blob/main/start.sh.
